### PR TITLE
[12.0][FIX] mrp_subcontracting: Overwrite active_id context in create()

### LINKED ISSUE #526: [12.0] mrp_subcontracting: picking cannot be validated
Steps to reproduce:
1. Create a subcontract BoM.
image
2. Create a purchase order with the product above and confirm the order.
3. Make sure there are more stock.picking/purchase.order records than mrp.production records.
4. Browse the picking from the purchase order / from the Inventory menu and validate the picking.

Current behavior:
"Record does not exist or has been deleted." is raised and the picking cannot be confirmed.

From Purchase Order:
image

From Inventory menu:
image

### PR DESCRIPTION
**Pull request by TimLai125**
Overwrite the active_id in context when creating the `mrp.product.produce` record in `action_done`
Fixes: #526

### CODE REVIEW

File: mrp_subcontracting/models/stock_picking.py [hunk 1]
```diff
@@ -72,8 +72,10 @@ def action_done(self):
                         })
                 else:
                     for move_line in move.move_line_ids:
+                        active_id = production and production.id or False
```
Comment (pedrobaeza) [inline mrp_subcontracting/models/stock_picking.py]: You can simplify this:

```suggestion
                        active_id = production.id
```

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
As this code is backported from Odoo, can this happen also there in v13?

**Review by marcelsavegnago — APPROVED** [review]
LGTM

**Review by bodedra — APPROVED** [review]

**Pull request merged**